app/ca.go

```diff
@@ -0,0 +1,117 @@
+package app
+
+import (
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+)
+
+type caCert struct {
+	Fingerprint       string    `json:"fingerprint"`
+	Issuer            string    `json:"issuer"`
+	Subject           string    `json:"subject"`
+	NotBefore         time.Time `json:"notBefore"`
+	NotAfter          time.Time `json:"notAfter"`
+	Base64EncodedData string    `json:"base64EncodedData"`
+}
+
+type caCerts []caCert
+
+func parseCertificates(encodedCerts string) (caCerts, error) {
+
+	// decode the cert bundle
+	bundle, err := base64.StdEncoding.DecodeString(encodedCerts)
+	if err != nil {
+		return nil, err
+	}
+	if len(bundle) == 0 {
+		return caCerts{}, nil
+	}
+
+	certBlocks := [][]byte{}
+	var blocks []byte
+	p := bundle
+	for {
+		var block *pem.Block
+		var rem []byte
+		block, rem = pem.Decode(p)
+		if block == nil {
+			break
+		}
+		blocks = append(blocks, block.Bytes...)
+		certBlocks = append(certBlocks, []byte(strings.TrimSpace(string(p[:len(p)-len(rem)]))))
+		p = rem
+	}
+	if len(blocks) == 0 {
+		return nil, fmt.Errorf("failed to decode certificates")
+	}
+	// If p is greater than 0, then this means that there was a portion of the certificate that
```

Comment:
the server actually will still save the bundle even if there are leftover bytes from the bundle. We might want to print a warning here vs actually failing

Reply:
The server will reject the bundle if it cannot parse it right? I believe we perform this exact check on the server.

Reply:
note that the server got is doing fmt.Println, not returning an error, whereas your code would return an error.

Reply:
Got it. 
I am wondering if we should be more strict on the cli, especially when they are adding and remove certs to not have malformed inputs. If they still want to push partial certs they can still use the `set` command.